blenderproc/python/writer/GifWriterUtility.py

```diff
@@ -0,0 +1,210 @@
+
+
+class GifWriterUtility:
+    
+    @staticmethod
+    def _provide_directory(output_dir_path: str) -> str:
+        """ Generates subdirectory for .gif files if not existent """
+        output_dir_path = os.path.join(output_dir_path, 'gif_animations')
+        if not os.path.exists(output_dir_path):
+            print('\n Generate output folder: ' + output_dir_path)
+            os.makedirs(output_dir_path)
+        return output_dir_path
+
+    @staticmethod
+    def _select_keys(
+            output_data_dict: Dict[str, List[Union[np.ndarray, list, dict]]]
+            ) -> List[str]:
+        """ Sorts out keys which are just metadata and not plottable """
+        keys_to_use = []
+        for key in output_data_dict.keys():
+            value = output_data_dict[key][0]
+            value = np.array(value)
+            if np.issubdtype(value.dtype, np.string_) or len(value.shape) == 1:
+                pass # metadata
+            else:
+                keys_to_use.append(key)
+        return keys_to_use
+
+    @staticmethod
+    def _cache_png(keys_to_use: List[str], 
+            output_data_dict: Dict[str, List[Union[np.ndarray, list, dict]]]
+            ) -> Dict[str, str]:
+        """
+        Builds temporary folders with .png collections 
+        and returns the locations as dictionary. 
+        """
+
+        if not set(keys_to_use) <= set(output_data_dict.keys()):
+            raise ValueError("The keys_to_use list must be contained in"
+                            "the list of keys from the output_data_dict!")
+
+        to_animate = dict()
+        for key in keys_to_use:
+            value = output_data_dict[key][0]
+            value = np.array(value)
+            # Check if frames are rendered in stereo vision
+            if value.shape[0] is 2:
+                # stereo images
+                for index, perspective in enumerate(['_L', '_R']):
+                    tmp_dir = tempfile.TemporaryDirectory()
+                    print('Temporary directory for ' 
+                            + key 
+                            + perspective 
+                            + ': ' 
+                            + tmp_dir.name)
+                    to_animate.update({key + perspective: tmp_dir})
+                    for number, frame in enumerate(output_data_dict[key]):
+                        vis_data(key=key,
+                                data=frame[index], 
+                                save_to_file=tmp_dir.name + f'/{number}.png')
+            else:
+                # non stereo images
+                tmp_dir = tempfile.TemporaryDirectory()
+                print('Temporary directory for ' + key + ': ' + tmp_dir.name)
+                to_animate.update({key: tmp_dir})
+                for number, frame in enumerate(output_data_dict[key]):
+                    vis_data(key=key, 
+                            data=frame, 
+                            save_to_file=tmp_dir.name + f'/{number}.png')
```

Comment:
Same here. No merging of paths without a library (Windows)

Reply:
Done. Like you said.